packages/features/eventtypes/components/MultiplePrivateLinksController.tsx

```diff
@@ -1,97 +1,391 @@
+                          data-testid={`remove-private-link-${originalIndex}`}
+                          variant="icon"
+                          type="button"
+                          StartIcon="trash-2"
+                          color="destructive"
+                          className="ml-1 border-none"
+                          onClick={() => removePrivateLink(originalIndex)}
+                        />
+                      </div>
+                    </div>
+                    <div data-testid="private-link-description" className="mt-1 text-sm text-gray-500">
+                      {linkDescription}
+                    </div>
                   </li>
                 );
               })}
+              <Button
+                color="minimal"
+                StartIcon="plus"
+                onClick={addPrivateLink}
+                data-testid="add-private-link-button">
+                {t("add_a_multiple_private_link")}
+              </Button>
+            </ul>
+          );
+        }}
+      />
+
+      <Dialog open={isDialogOpen} onOpenChange={setIsDialogOpen}>
+        <DialogContent title={t("link_settings")} type="creation">
+          <div className="mb-4 space-y-4">
+            <RadioArea.Group
+              className="space-y-2"
+              value={selectedType}
+              data-testid="private-link-radio-group"
+              onValueChange={(value: "time" | "usage") => {
+                setSelectedType(value);
+                updateLinkSettings(currentLinkIndex, value);
+              }}>
+              <RadioArea.Item value="usage" data-testid="private-link-usage" className="w-full text-sm">
+                <Label className="mb-0 cursor-pointer text-sm font-medium">
+                  {t("usage_based_expiration")}
+                </Label>
+                {selectedType === "usage" && (
+                  <div className="mt-2 w-[180px]">
+                    <NumberInput
+                      required
+                      min={1}
+                      data-testid="private-link-usage-count"
+                      placeholder={t("number_of_uses")}
+                      value={maxUsageCount === null ? "" : maxUsageCount}
+                      onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+                        const value = e.target.value === "" ? null : parseInt(e.target.value);
+                        if (e.target.value === "") {
+                          setMaxUsageCount(null);
+                        } else if (!isNaN(Number(value)) && Number(value) > 0) {
+                          setMaxUsageCount(value);
+                          updateLinkSettings(currentLinkIndex, "usage", undefined, value);
+                        }
+                      }}
+                    />
+                  </div>
+                )}
+              </RadioArea.Item>
+              <RadioArea.Item data-testid="private-link-time" value="time" className="w-full text-sm">
```

Comment:
Not sure if that's an issue with our existing component but I am having issues when clicking on the text and not directly on the radio button. Sometimes it works and the button also highlights when hovering over the text

<img width="545" alt="Screenshot 2025-07-10 at 10 32 58 AM" src="https://github.com/user-attachments/assets/e5d967c6-ede1-4de7-aadb-1bf41a36d997" />

Reply:
Fixed